File: php/t/format_speed.php
<?php

include_once('../Plurr.php');

$p = new Plurr();
$s = 'Do you want to delete {N_PLURAL:this {N} file|these {N} files} permanently?';
$params = array('N' => 5);
$x = 100000;

$start = microtime(true);

for ($i = 0; $i < $x; $i++) {
  $dummy = $p->format($s, $params);
}

$end = microtime(true);
$time = $end - $start;

print("Execution time ($x calls): $time sec (" . ($time * 1000 / $x) . ' ms per call)');

?>

File: perl/t/format_speed.pl
#!/usr/bin/perl

use strict;

BEGIN {
  use File::Spec::Functions qw(rel2abs);
  use File::Basename;
  unshift(@INC, dirname(rel2abs($0)).'/..');
}

use Time::HiRes qw(gettimeofday tv_interval);

use Plurr;

my $p = Plurr->new();
my $s = 'Do you want to delete {N_PLURAL:this {N} file|these {N} files} permanently?';
my $params = {'N' => 5};
my $x = 100000;

my $start = [gettimeofday];

for (my $i = 0; $i < $x; $i++) {
  my $dummy = $p->format($s, $params);
}

my $end = [gettimeofday];
my $time = tv_interval($start, $end);
print("Execution time ($x calls): $time sec (" . ($time * 1000 / $x) . ' ms per call)');
